### PR TITLE
NEXT-12789 - Administration: Handling media

### PR DESCRIPTION
**Pull request by NiklasLimberg**
## What should be done?

Create a new article on our [GitBook instance|https://app.gitbook.com/@shopware/s/shopware/] which explains how to handle medias in the administration.

This article should mention:
- The prerequisite, a working plugin (Refer to the plugin base guide)
- Every other prerequisite you figure out during writing the guide (e.g. a subscriber, knowing how to create a service, a controller, etc.)
- A short code example, including an explanation, on how to do it
- For example: How to load (and show) a media, how to implement a media picker or smth like this

Category: Extensions > Plugins > Administration

## Are there already guides in the previous documentation for this?
Unfortunately, No. But please have a look at our [example guide|https://app.gitbook.com/@shopware/s/shopware/guides/plugins/plugins/plugin-base-guide](e.g. note the PLACEHOLDERs for cross-references, that do not exist yet) and the [Writing guidelines|https://app.gitbook.com/@shopware/s/shopware/resources/guidelines/documentation/writing].

For more information, ask me, [~p.stahl].

### CODE REVIEW

File: guides/plugins/plugins/administration/handling-media.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+This is done through the `sw-media-upload-v2` component as seen below:
+
+```html
+<div>
+    <sw-media-upload-v2
+        uploadTag="my-upload-tag"
+        :allowMultiSelect="false"
+        variant="regular"
+        :autoUpload="true"
+		label="My image-upload">
```
Comment (raknison) [inline guides/plugins/plugins/administration/handling-media.md:15]:
```suggestion
        :autoUpload="true"
        label="My image-upload">
```

File: guides/plugins/plugins/administration/handling-media.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    </sw-media-upload-v2>
+</div>
+```
+
+As you saw in the code sample, the `sw-media-upload-v2` is pretty configurable trough props.
```
Comment (raknison) [inline guides/plugins/plugins/administration/handling-media.md:20]:
```suggestion
As you can see in the code sample below, the `sw-media-upload-v2` is pretty configurable through props.
```

just my taste

File: guides/plugins/plugins/administration/handling-media.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+The `sw-upload-listener` component, can as seen below, be used in conjunction with an `sw-media-upload-v2` component.
+
+```html
+<div>
+    <sw-media-upload-v2
+        uploadTag="my-upload-tag"
+        :allowMultiSelect="false"
+        variant="regular"
+		label="My image-upload">
```
Comment (raknison) [inline guides/plugins/plugins/administration/handling-media.md:46]:
```suggestion
        variant="regular"
        label="My image-upload">
```

**Review by raknison — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/administration/handling-media.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Handling media
+
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/handling-media.md:2]: Please write a little Overview of the guide as introduction.

File: guides/plugins/plugins/administration/handling-media.md (hunk 2 above)
Comment (leichteckig) [reply guides/plugins/plugins/administration/handling-media.md:20]: Should we use the word `properties`, as long version?

File: guides/plugins/plugins/administration/handling-media.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+| `label`            | The text that is displayed in the header                                                                                        |
+| `defaultFolder`    | The path where the file will be put                                                                                             |
+| `targetFolderId`   | The `targetFolderId` that will be used as a backup to the `defaultFolder`                                                       |
+| `helpText`         | Sets the `helpText` displayed in the header of the component                                                                    |
+| `fileAccept`       | Sets what the underlying <input>, accepts standard is `image/*`                                                                 |
+| `disabled`         | Disables the whole component                                                                                                    |
+
+## Keeping track of uploads
+
+The `sw-upload-listener` component, can as seen below, be used in conjunction with an `sw-media-upload-v2` component.
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/handling-media.md:38]:
```suggestion
As seen below, the `sw-upload-listener` component can be used in conjunction with an `sw-media-upload-v2` component.
```

File: guides/plugins/plugins/administration/handling-media.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+Media can be previewed with the `sw-media-preview-v2` component as seen below:
+
+```html
+<sw-media-preview-v2
+    :source="some-id">
+</sw-media-preview-v2>
+```
+
+As previously mention this component is already embedded within the `sw-media-upload-v2`.
+But using it as a separate component you get access to the following configuration options:
```
Comment (leichteckig) [inline guides/plugins/plugins/administration/handling-media.md:77]:
```suggestion
However, using it as a separate component you get access to the following configuration options:
```

**Review by leichteckig — CHANGES_REQUESTED** [review]

**Comment by JoshuaBehrens** [comment]
Will this replace the component-library?

**Comment by NiklasLimberg** [comment]
No it won't. But I didn't know how to approach this differently

**Comment by lernhart** [comment]
> Will this replace the component-library?

No it won't. Unfortunately, we can not link the component library here, as the component library's sub pages can not be visited via direct link. In the current setup, you'll get a 403 trying to visit any other page than the landing page.

File: guides/plugins/plugins/administration/handling-media.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+```
+
+Notice that the `uploadTag` needs to be the same in the `sw-media-upload-v2` and the `sw-upload-listener` for them to communicate properly.
+Beyond the `media-upload-finish` event there are a few more events:
+
+| Event                 | Description                                        |
+|-----------------------|----------------------------------------------------|
+| `media-upload-add`    | This event is triggered when an upload is added    |
+| `media-upload-finish` | This event is triggered when an upload finishes    |
+| `media-upload-fail`   | This is triggered on an upload failing             |
```
Comment (lernhart) [inline guides/plugins/plugins/administration/handling-media.md:62]:
```suggestion
| `media-upload-fail`   | This event is triggered on an upload failing       |
```

File: guides/plugins/plugins/administration/handling-media.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+Media can be previewed with the `sw-media-preview-v2` component as seen below:
+
+```html
+<sw-media-preview-v2
+    :source="some-id">
+</sw-media-preview-v2>
+```
+
+As previously mention this component is already embedded within the `sw-media-upload-v2`.
```
Comment (lernhart) [inline guides/plugins/plugins/administration/handling-media.md:76]:
```suggestion
As previously mentioned this component is already embedded within the `sw-media-upload-v2`.
```

File: guides/plugins/plugins/administration/handling-media.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+However, using it as a separate component you get access to the following configuration options:
```
Comment (lernhart) [inline guides/plugins/plugins/administration/handling-media.md:77]:
```suggestion
However, using it as a separate component gets you access to the following configuration options:
```

File: guides/plugins/plugins/administration/handling-media.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+| Property         | Function                                                                            |
+|------------------|-------------------------------------------------------------------------------------|
+| `source`         | The `id` or alternately the path to the media to be previewed                       |
+| `showControls`   | Controls whether media such as videos or audio shows controls                       |
+| `autoplay`       | Controls whether media such as videos or audio autoplays                            |
+| `hideTooltip`    | Hides the tooltip witch the filename of the media in at the bottom of the component |
```
Comment (lernhart) [inline guides/plugins/plugins/administration/handling-media.md:84]:
```suggestion
| `hideTooltip`    | Hides the the filename tooltip of the media in at the bottom of the component       |
```

**Review by lernhart — CHANGES_REQUESTED** [review]

**Comment by JoshuaBehrens** [comment]
Unfortunate that the component library does not get a chance to be referenced here. Can someone put it on the list? The component library is a real nice docs for this

**Review by lernhart — CHANGES_REQUESTED** [review]
Please resolve the issues again

**Review by lernhart — APPROVED** [review]

**Review by leichteckig — CHANGES_REQUESTED** [review]
One last thing :)

**Review by raknison — APPROVED** [review]

**Review by leichteckig — APPROVED** [review]

**Pull request merged**